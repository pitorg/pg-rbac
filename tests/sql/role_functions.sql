SELECT rbac.add_role('admin') AS role_id \gset
SELECT rbac.add_role('admin'); -- error since already added
SELECT rbac.delete_role(:'role_id'::UUID);
SELECT rbac.delete_role(:'role_id'::UUID); -- NULL, already deleted
