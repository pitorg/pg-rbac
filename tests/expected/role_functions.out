SELECT rbac.add_role('admin') AS role_id \gset
SELECT rbac.add_role('admin'); -- error since already added
ERROR:  duplicate key value violates unique constraint "roles_name_key"
DETAIL:  Key (name)=(admin) already exists.
CONTEXT:  SQL function "add_role" statement 1
SELECT rbac.delete_role(:'role_id'::UUID);
 delete_role 
-------------
 t
(1 row)

SELECT rbac.delete_role(:'role_id'::UUID); -- NULL, already deleted
 delete_role 
-------------
 
(1 row)

